checks/store/legacyrank/legacy_rank.go

```diff
@@ -0,0 +1,99 @@
+package legacyrank
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/csv"
+	"errors"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+var ErrNotFound = errors.New("domain not found")
+
+type Getter interface {
+	GetLegacyRank(domain string) (int, error)
+}
+
+type GetterFunc func(domain string) (int, error)
+
+func (f GetterFunc) GetLegacyRank(domain string) (int, error) {
+	return f(domain)
+}
+
+type InMemoryStore struct{}
+
+var once sync.Once
+var data map[string]int //map of domain to rank
+
+func NewInMemoryStore() *InMemoryStore {
+	return &InMemoryStore{}
+}
+
+func (s *InMemoryStore) GetLegacyRank(url string) (int, error) {
+	once.Do(func() {
+		var err error
+		data, err = load()
+		if err != nil {
+			log.Println(err)
+		}
+	})
+
+	rank, ok := data[url]
+	if !ok {
+		return -1, ErrNotFound
+	}
+	return rank, nil
+}
+
+func load() (map[string]int, error) {
```

Comment:
This might be able to be split up a bit smaller

Reply:
yeah i dont like the the fact it needs to be loaded into memory first, difference of taking in a Reader and ReaderAt with the zip reader, its possible to stream a zip file. but not with the native APIs